essim/src/ssim_cpu.c

```diff
@@ -144,11 +144,17 @@ static eCPUType QueryCpuType() {
 
 static eCPUType QueryCpuType() { return cpu_neon; }
 
+#if ARM_BUILD_FIXES
+#else
+
+static eCPUType QueryCpuType() { return cpu_neon; }
+#endif
+
 #endif /* defined(__ANDROID__) */
 
 #else /* an unknown platform */
 
-static eCPUType QueryCpuType() { return cpuPlain; }
+static eCPUType QueryCpuType() { return cpu_neon; }
```

Comment:
this doesn't seem right, why would this default to `cpu_neon` for unknown platforms?

Reply:
This issue has taken by considering "__LINUX", precompilation macro.  Default is reverted to "cpuPlain".

Reply:
this nested set of conditions is a bit hard to read, so I might be reading this wrong, but it still doesn't feel right. This definition of `cpu_neon` is outside of the `#elif defined(_ARM) || defined(_ARM64)` block. If neither of those is true then we shouldn't default to neon afaict. Within the checks for arm if we want to default to neon, that seems fine.